File: 0463-island-perimeter/0463-island-perimeter.java
class Solution {
    public int islandPerimeter(int[][] grid) {
        int[] dr = {0, 0, -1, 1};
        int[] dc = {1, -1, 0, 0};
        int n = grid.length, m = grid[0].length;
        int result = 0;

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                if (grid[i][j] == 1) {
                    int cnt = 0;
                    for (int k = 0; k < 4; k++) {
                        int r = i + dr[k], c = j + dc[k];
                        if (r < 0 || r >= n || c < 0 || c >= m) {
                            cnt++;
                            continue;
                        }
                        if (grid[r][c] == 0) {
                            cnt++;
                        }
                    }
                    result += cnt;
                }
            }
        }

        return result;
    }
}

File: 0463-island-perimeter/0463-island-perimeter.py
class Solution:
    def islandPerimeter(self, grid: List[List[int]]) -> int:
        dr = (0, 0, -1, 1)
        dc = (1, -1, 0, 0)
        n = len(grid)
        m = len(grid[0])
        result = 0

        for i in range(n):
            for j in range(m):
                if grid[i][j] == 1:
                    cnt = 0
                    for k in range(4):
                        r, c = i + dr[k], j + dc[k]
                        if r < 0 or r >= n or c < 0 or c >= m:
                            cnt += 1
                            continue
                        if grid[r][c] == 0:
                            cnt += 1
                    result += cnt

        return result
